src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c

```diff
@@ -2531,8 +2531,21 @@ exec_mark_cmd_state(struct exec_core *exec, struct xocl_cmd *xcmd, enum ert_cmd_
 static inline void
 exec_mark_cmd_complete(struct exec_core *exec, struct xocl_cmd *xcmd)
 {
+	//For soft kernels get cmd state and return_code
+	uint32_t cmd_state = ERT_CMD_STATE_COMPLETED;
+	if (cmd_type(xcmd) == ERT_SCU) {
+		struct xocl_ert *xert = exec_is_ert(exec) ? exec->ert : NULL;
+		if (xert) {
+			uint32_t slot_addr = xcmd->slot_idx * xert->slot_size;
+			struct ert_start_kernel_cmd *pkt = xcmd->ert_cu;
+			xocl_memcpy_fromio((void*)&pkt->return_code, xert->cq_base + slot_addr, 2 * sizeof(u32));
```

Comment:
why 2 4-bytes? thought we just need to copy the return code

Reply:
cmd state & return code are copied. state can be crashed or error_code or completed